sub-packages/bionemo-esm2/src/bionemo/esm2/run/recipes.py

```diff
@@ -0,0 +1,447 @@
+        num_dataset_workers=8,
+        train_cluster_path=args.train_cluster_path,
+        train_database_path=args.train_database_path,
+        valid_cluster_path=args.valid_cluster_path,
+        valid_database_path=args.valid_database_path,
+    )
+
+    return MainConfig(
+        data_config=data_config,
+        parallel_config=esm2_base_parallel_config(),
+        training_config=esm2_base_training_config(),  # no changes for 8m
+        bionemo_model_config=esm2_650m_model_config(args.initial_ckpt_path),
+        optim_config=esm2_base_optimizer_scheduler_config(),  # no changes for 8m
+        experiment_config=esm2_650m_experiment_config(args.result_dir),
+        wandb_config=esm2_650m_wandb_config(),
+    )
+
+
+def esm2_3b_parallel_config() -> ParallelConfig:
+    """Parallel config for ESM2 3b."""
+    return ParallelConfig(
+        tensor_model_parallel_size=2,
+        pipeline_model_parallel_size=1,
+        # TODO: is this correct?
+        accumulate_grad_batches=1,
+        ddp="megatron",
+        # NOTE assumes 8xGPU node. Can always edit the config.
+        num_devices=8,
+    )
+
+
+def esm2_3b_model_config(initial_ckpt_path=None) -> ExposedESM2PretrainConfig:
+    """Model config for ESM2 3b."""
+    return ExposedESM2PretrainConfig(
+        num_layers=36,
+        hidden_size=2560,
+        ffn_hidden_size=2560 * 4,
+        num_attention_heads=40,
+        seq_length=1024,
+        biobert_spec_option=BiobertSpecOption.esm2_bert_layer_with_transformer_engine_spec,
+        initial_ckpt_path=initial_ckpt_path,
+        params_dtype="bf16-mixed",
+        pipeline_dtype="bf16-mixed",
+        autocast_dtype="bf16-mixed",
+    )
+
+
+def esm2_3b_wandb_config() -> WandbConfig:
+    """Wandb config for ESM2 3b."""
+    return WandbConfig(
+        entity="esm2-3b_pretraining",
+        project="esm2-3b_pretraining",
+        group="esm2-3b",
+        tags=["esm2-3b"],
+        offline=True,
+        anonymous=True,
+        id="1",
+        log_model=False,
+    )
+
+
+def esm2_3b_recipe(args) -> MainConfig[ExposedESM2PretrainConfig, ESM2DataConfig]:
+    """Recipe for ESM2 3b."""
+    data_config = ESM2DataConfig(
+        min_seq_length=1024,
+        max_seq_length=1024,
+        micro_batch_size=1,
+        num_dataset_workers=8,
+        train_cluster_path=args.train_cluster_path,
+        train_database_path=args.train_database_path,
+        valid_cluster_path=args.valid_cluster_path,
+        valid_database_path=args.valid_database_path,
+    )
+
+    return MainConfig(
+        data_config=data_config,
+        parallel_config=esm2_3b_parallel_config(),
+        training_config=esm2_base_training_config(),  # no changes for 8m
+        bionemo_model_config=esm2_3b_model_config(args.initial_ckpt_path),
+        optim_config=esm2_base_optimizer_scheduler_config(),  # no changes for 8m
+        experiment_config=esm2_650m_experiment_config(args.result_dir),
+        wandb_config=esm2_3b_wandb_config(),
+    )
```

Comment:
@farhadrgh @sichu2023  do these look accurate?

Reply:
Similar to above.